extensions/amp-story/1.0/amp-story-system-layer.js

```diff
@@ -548,29 +553,31 @@ export class SystemLayer {
   }
 
   /**
-   * Hides audio message after elapsed time.
+   * Hides message after elapsed time.
+   * @param {string} message
    * @private
    */
-  hideAudioMessageAfterTimeout_() {
+  hideMessageAfterTimeout_(message) {
     if (this.timeoutId_) {
       this.timer_.cancel(this.timeoutId_);
     }
     this.timeoutId_ = this.timer_.delay(
-      () => this.hideAudioMessageInternal_(),
-      HIDE_AUDIO_MESSAGE_TIMEOUT_MS
+      () => this.hideMessageInternal_(message),
+      HIDE_MESSAGE_TIMEOUT_MS
     );
   }
 
   /**
-   * Hides audio message.
+   * Hides message.
+   * @param {string} message
    * @private
    */
-  hideAudioMessageInternal_() {
+  hideMessageInternal_(message) {
     if (!this.isBuilt_) {
       return;
     }
     this.vsync_.mutate(() => {
-      this.getShadowRoot().setAttribute(MESSAGE_DISPLAY_CLASS, 'noshow');
+      this.getShadowRoot().setAttribute(message, 'noshow');
```

Comment:
Sorry I know this is the type of thing that's very difficult to test (manually, or in any automated fashion), but how do multiple messages override each other?  Does the first one just disgust completely once the second one comes in?

I suspect that would be the case, but I'm not actually sure whether that's okay?  This can result in weird edge cases when one message is triggered right after another.  It's probably fine if it's two audio message since they'd both come from user interactions, but if it's an audio message followed by a live list update, the user won't have had time to read the audio message.

Should we consider queuing the messages like toast frameworks (likely) would?

Reply:
I tested it and right now they are both shown, and eventually disappear. Since this seems to be a very specific edge case, do you mind if I address it in a follow-up?

I filed #22778 to keep track.